gui/wxpython/lmgr/frame.py

```diff
@@ -1672,121 +1838,30 @@ def OnShowAttributeTable(self, event, selection=None):
         # show ATM window
         dbmanager.Show()
 
-    def OnNewDisplay(self, event=None):
-        """Create new layer tree and map display instance"""
-        self.NewDisplay()
-
-    def NewDisplay(self, name=None, show=True):
-        """Create new layer tree, which will
-        create an associated map display frame
-
-        :param name: name of new map display
-        :param show: show map display window if True
-
-        :return: reference to mapdisplay intance
-        """
-        Debug.msg(1, "GMFrame.NewDisplay(): idx=%d" % self.displayIndex)
-
-        # make a new page in the bookcontrol for the layer tree (on page 0 of
-        # the notebook)
-        self.pg_panel = wx.Panel(self.notebookLayers, id=wx.ID_ANY, style=wx.EXPAND)
-        if name:
-            dispName = name
+    def _onMapDisplayClose(self, page, askIfSaveWorkspace=True):
+        """Closes Map Display window and its associated layer tree page."""
+
+        def CanCloseDisplay():
+          """Check if user wants to close display"""
+          pgnum = self.notebookLayers.GetPageIndex(page)
+          name = self.notebookLayers.GetPageText(pgnum)
+          caption = _("Close Map Display {}").format(name)
+          if not askIfSaveWorkspace or (
+              askIfSaveWorkspace and self.workspace_manager.CanClosePage(caption)
+          ):
+              return pgnum
+          return None
+
+        if CanCloseDisplay():
+            pgnum = CanCloseDisplay()
+            if pgnum is not None:
+                self.GetMapDisplay().CleanUp()
+                if pgnum > -1:
+                    self._closePageNoEvent(page_index=pgnum)
+                    # Destroy is called when notebook page is deleted
```

Comment:
The nested function is not needed here anymore, so I suggest just putting the code in the main body of the function. Also use the parameter `page`, don't use GetMapDisplay,  because that refers to the current one, not necessarily the one connected to `page`.

Reply:
Also these changes in closing will likely impact standalone displays (d.mon, g.guiv/rdigit).